modules/tts/tts-adapters/tts-adapter-sapinative/src/main/java/org/daisy/pipeline/tts/sapi/impl/SAPIService.java

```diff
@@ -0,0 +1,173 @@
+package org.daisy.pipeline.tts.sapi.impl;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.util.Map;
+
+import javax.naming.directory.InvalidAttributeValueException;
+import javax.sound.sampled.AudioFormat;
+
+import org.daisy.common.file.URLs;
+import org.daisy.pipeline.tts.onecore.Onecore;
+import org.daisy.pipeline.tts.onecore.OnecoreResult;
+import org.daisy.pipeline.tts.onecore.SAPI;
+import org.daisy.pipeline.tts.onecore.SAPIResult;
+import org.daisy.pipeline.tts.TTSEngine;
+import org.daisy.pipeline.tts.TTSService;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(
+	name = "sapi-onecore-tts-service",
+	service = { TTSService.class }
+)
+public class SAPIService implements TTSService {
+
+	private static final Logger Logger = LoggerFactory.getLogger(SAPIEngine.class);
+
+	private boolean onecoreIsLoaded = false;
+	private boolean sapiIsLoaded = false;
+
+	private AudioFormat sapiAudioFormat = null;
+
+	@Override
+	public TTSEngine newEngine(Map<String, String> params) throws Throwable {
+		int priority = convertToInt(params, "org.daisy.pipeline.tts.sapi.priority", 7);
+		synchronized (this) {
+			// try to load both sapi and onecore to keep using third party voices that could have
+			// been installed on sapi registry and not exposed to the onecore registry
+			try {
+				if (!onecoreIsLoaded) {
+					loadOnecoreDLL();
+					onecoreIsLoaded = true;
+				}
+				int res = Onecore.initialize();
+				if (res != OnecoreResult.SAPINATIVE_OK.value()) {
+					onecoreIsLoaded = false;
+					Onecore.dispose();
+					throw new SynthesisException(
+							"Onecore initialization failed with error code '" + res + "': " + OnecoreResult.valueOfCode(res));
+				}
+			} catch (Exception e) {
+				Logger.warn(e.getMessage());
+			}
+			try {
+				int sapiSampleRate = convertToInt(params, "org.daisy.pipeline.tts.sapi.samplerate", 22050);
+				int sapiBytesPerSample = convertToInt(params, "org.daisy.pipeline.tts.sapi.bytespersample", 2);
```

Comment:
These settings are used for legacy SAPI voices, but not for OneCore voices. Why was this again? (You possibly already explained but I forget.) Should this be mentioned in the [documentation](https://github.com/daisy/pipeline-modules/blob/master/tts/tts-common/doc/index.md#sapi)?

Reply:
For OneCore voice, you cannot change the sample rate and bytes per sample settings of the synthesizer anymore.
(Or rather I have not found any settings that would allow to change them on the WinRT API pages or from the documentation exposed when working in visual studio with c++/winrt)

It might be indeed useful to mention this in the documentation that list the options for TTS engines.